### PR TITLE
Implement hasTag onto BiomeContext

### LINKED ISSUE #271: BiomeSelectionContext wrapper does not expose necessary information
The BiomeModification API currently does not expose many methods from the underlying Fabric `BiomeSelectionContext` that are necessary for the API's use. For example, I wish to filter a feature by a biome tag, but this is not possible because it requires registry access and the modifications API calls its checks before the server is fully constructed and thus accessible for registry access.

The API should expose `boolean hasTag(TagKey<Biome> tag);` or `getHolder` via `BiomeContext`, or even the `Biome` instance itself. As of current `BiomeContext` does not include enough data for a lot of things.

### PR DESCRIPTION
**Pull request by itsmeow**
Untested, probably works. Really hacky solution for forge, could be improved?

Resolves #271

### CODE REVIEW

**Comment by shedaniel** [comment]
was this tested?

**Comment by itsmeow** [comment]
> Untested

**Comment by shedaniel** [comment]
yeah do that 😅

**Comment by itsmeow** [comment]
Tested it, seems to work on both implementations.

File: forge/src/main/java/dev/architectury/registry/level/biome/forge/BiomeModificationsImpl.java [hunk 1]
```diff
@@ -136,6 +139,11 @@ public Optional<ResourceLocation> getKey() {
             public BiomeProperties getProperties() {
                 return properties;
             }
+            
+            @Override
+            public boolean hasTag(TagKey<Biome> tag) {
+                return GameInstance.getServer().registryAccess().registryOrThrow(Registry.BIOME_REGISTRY).getHolder(biomeResourceKey.get()).get().is(tag);
```
Comment (shedaniel) [inline forge/src/main/java/dev/architectury/registry/level/biome/forge/BiomeModificationsImpl.java]: Would it be wise here to check if the `getHolder` optional is null?

Comment (itsmeow) [reply forge/src/main/java/dev/architectury/registry/level/biome/forge/BiomeModificationsImpl.java]: Maybe, but it'll error out anyway, so I don't see the point. The chances of it being null are near zero.

**Comment by itsmeow** [comment]
Should be good to merge now.

**Review by shedaniel — APPROVED** [review]
Some styling issues, but we will fix that on merge, LGTM

**Pull request merged**